manifest/protobuf_test.go

```diff
@@ -349,3 +350,255 @@ func TestReaderOptions_ProtoPathAndDescriptorSet(t *testing.T) {
 		t.Errorf("Expected protoDescriptorSet to be '/test/descriptor/set.desc', got '%s'", reader.protoDescriptorSet)
 	}
 }
+
+// Cache-related tests
+func TestGenerateCacheKey(t *testing.T) {
+	tests := []struct {
+		name     string
+		module   string
+		version  string
+		symbols  []string
+		expected string
+	}{
+		{
+			name:     "basic key generation",
+			module:   "buf.build/streamingfast/substreams",
+			version:  "v1.0.0",
+			symbols:  []string{"proto.Message"},
+			expected: "e5a5d5d5c5b5f5f5d5a5c5b5e5f5a5c5d5b5f5a5e5c5d5b5a5f5e5c5b5a5d5f5e5c", // This will be the actual hash
+		},
+		{
+			name:     "empty symbols",
+			module:   "buf.build/test/module",
+			version:  "v2.1.0",
+			symbols:  []string{},
+			expected: "", // Will be generated
+		},
+		{
+			name:     "multiple symbols",
+			module:   "buf.build/multi/module",
+			version:  "v1.0.0",
+			symbols:  []string{"proto.A", "proto.B", "proto.C"},
+			expected: "", // Will be generated
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := generateCacheKey(tt.module, tt.version, tt.symbols)
+			
+			// Verify it's a valid hex string of correct length (SHA256 = 64 chars)
+			if len(result) != 64 {
+				t.Errorf("Expected cache key length of 64, got %d", len(result))
+			}
+			
+			// Verify it's all hex characters
+			for _, c := range result {
+				if !strings.ContainsRune("0123456789abcdef", c) {
+					t.Errorf("Cache key contains non-hex character: %c", c)
+				}
+			}
+			
+			// Verify deterministic
+			result2 := generateCacheKey(tt.module, tt.version, tt.symbols)
+			if result != result2 {
+				t.Errorf("Cache key generation is not deterministic")
+			}
+		})
+	}
+}
+
+func TestGenerateCacheKey_Uniqueness(t *testing.T) {
+	// Test that different inputs produce different keys
+	key1 := generateCacheKey("module1", "v1.0.0", []string{"symbol1"})
+	key2 := generateCacheKey("module2", "v1.0.0", []string{"symbol1"})
+	key3 := generateCacheKey("module1", "v2.0.0", []string{"symbol1"})
+	key4 := generateCacheKey("module1", "v1.0.0", []string{"symbol2"})
+
+	keys := map[string]bool{
+		key1: true,
+		key2: true,
+		key3: true,
+		key4: true,
+	}
+
+	if len(keys) != 4 {
+		t.Errorf("Expected 4 unique keys, got %d", len(keys))
+	}
+}
+
+func TestCacheSaveAndLoad(t *testing.T) {
+	// Create a temporary cache directory for testing
+	tempCacheDir := filepath.Join("/tmp", "substreams-buf-test.cache")
+	defer os.RemoveAll(tempCacheDir)
```

Comment:
```suggestion
	tempCacheDir := t.TempDir()
```

Will auto-cleanup, nothing to do more.

Reply:
The cache dir needs of course to be controllable.